### DISCUSSION TITLE
Support for runtime namespace updates?

### DISCUSSION THREAD

**Discussion by quinlin-riggs** [eclipse-milo/milo#1143] [General]
In the example server provided in the repo the main server object is created then the example namespace is created/started and finally the server object is started.  

I am wondering if milo supports (or in theory supports) updating/modifying server namespaces at runtime (modifying existing namespaces or adding/removing a new namespace with lifecycle after some action is executed) in some way without needing to explicitly stop the server and recreate it.

**Comment by kevinherron** [comment]
I don't see why it wouldn't be supported, in theory. Starting up or shutting down a Namespace is really just registering/unregistering it with the server's AddressSpaceManager and registering a URI with the NamespaceTable.

It looks like the only thing unsupported about it might be that NamespaceTable doesn't have a method for removing a URI.

**Reply by kevinherron** [reply]
Removing a URI is a bit complicated... because you can't have a "gap" or null entry in the array, so if you removed anything other than the most recently registered URI the namespace array would be in an invalid state, or you would have to shift all further URIs down one index. Messy...

edit: (and you can't go shifting all the indices, at least while there are sessions open, because it would break any index mapping the client had...)

**Reply by quinlin-riggs** [reply]
Ah that makes sense. So in theory yes but there are some gotchas to consider - Thanks for the insight and the speedy reply!
